ton_client/src/debot/routines.rs

```diff
@@ -1,24 +1,79 @@
 use super::dengine::TonClient;
-use crate::crypto::{signing_box_sign, ParamsOfSigningBoxSign, SigningBoxHandle};
-use crate::net::{query_collection, ParamsOfQueryCollection};
+use crate::crypto::{
+    generate_random_bytes, nacl_box_keypair_from_secret_key, signing_box_sign, KeyPair,
+    ParamsOfGenerateRandomBytes, ParamsOfNaclBox, ParamsOfNaclBoxKeyPairFromSecret,
+    ParamsOfSigningBoxSign, SigningBoxHandle,
+};
 use crate::encoding::{decode_abi_bigint, decode_abi_number};
+use crate::net::{query_collection, ParamsOfQueryCollection};
 use chrono::{Local, TimeZone};
 
 pub async fn call_routine(
     ton: TonClient,
     name: &str,
     arg: &str,
     signer: Option<SigningBoxHandle>,
-) -> Result<String, String> {
+) -> Result<serde_json::Value, String> {
+    let arg_json: Result<serde_json::Value, String> =
+        serde_json::from_str(arg).map_err(|e| format!("argument is invalid json: {}", e));
     match name {
-        "convertTokens" => convert_string_to_tokens(ton, arg),
-        "getBalance" => get_balance(ton, arg).await,
-        "loadBocFromFile" => load_boc_from_file(ton, arg),
+        "convertTokens" => {
+            debug!("convertTokens({})", arg);
```

Comment:
All other matches check json parsing for error. Shouldn't check be here too?

Reply:
No, it takes raw string